src/gadgets/nonnative/bignat.rs

```diff
@@ -444,7 +444,11 @@ impl<Scalar: PrimeField> BigNat<Scalar> {
     let carry_bits = (((max_word.to_f64().unwrap() * 2.0).log2() - self.params.limb_width as f64)
       .ceil()
       + 0.1) as usize;
```

Comment:
This seems like it could lead to unexpected results in some situations due to a potential loss of precision with floats.